core/src/main/java/com/linecorp/armeria/common/HttpResponseBuilder.java

```diff
@@ -0,0 +1,270 @@
+    @FormatMethod
+    public HttpResponseBuilder content(@FormatString String format, Object... content) {
+        return (HttpResponseBuilder) super.content(format, content);
+    }
+
+    /**
+     * Sets the content for this response. The {@code content} is formatted by
+     * {@link String#format(Locale, String, Object...)} with {@linkplain Locale#ENGLISH English locale}.
+     */
+    @Override
+    @FormatMethod
+    public HttpResponseBuilder content(MediaType contentType, @FormatString String format,
+                                       Object... content) {
+        return (HttpResponseBuilder) super.content(contentType, format, content);
+    }
+
+    /**
+     * Sets the content for this response. The {@code content} will be wrapped using
+     * {@link HttpData#wrap(byte[])}, so any changes made to {@code content} will be reflected in the response.
+     */
+    @Override
+    public HttpResponseBuilder content(MediaType contentType, byte[] content) {
+        return (HttpResponseBuilder) super.content(contentType, content);
+    }
+
+    /**
+     * Sets the content for this response.
+     */
+    @Override
+    public HttpResponseBuilder content(MediaType contentType, HttpData content) {
+        return (HttpResponseBuilder) super.content(contentType, content);
+    }
+
+    /**
+     * Sets the {@link Publisher} for this response.
+     */
+    @Override
+    public HttpResponseBuilder content(MediaType contentType, Publisher<? extends HttpData> content) {
+        return (HttpResponseBuilder) super.content(contentType, content);
+    }
+
+    /**
+     * Sets the content for this response. The {@code content} that is converted into JSON
+     * using the default {@link ObjectMapper}.
+     */
+    @Override
+    public HttpResponseBuilder contentJson(Object content) {
+        return (HttpResponseBuilder) super.contentJson(content);
+    }
+
+    /**
+     * Sets a header for this response. For example:
+     * <pre>{@code
+     * HttpResponse.builder()
+     *             .ok()
+     *             .content("Hello, Armeria")
+     *             .header("Server", "foo")
+     *             .build();
+     * }</pre>
+     */
+    public HttpResponseBuilder header(CharSequence name, Object value) {
+        responseHeadersBuilder.setObject(requireNonNull(name, "name"),
+                                         requireNonNull(value, "value"));
+        return this;
+    }
+
+    /**
+     * Sets multiple headers for this response. For example:
+     * <pre>{@code
+     * HttpResponse.builder()
+     *             .ok()
+     *             .content("Hello, Armeria")
+     *             .headers(HttpHeaders.of("x-test-header", "foo", "Server", "baz"))
+     *             .build();
+     * }</pre>
+     * @see HttpHeaders
+     */
+    public HttpResponseBuilder headers(Iterable<? extends Entry<? extends CharSequence, String>> headers) {
+        requireNonNull(headers, "headers");
+        responseHeadersBuilder.set(headers);
```

Comment:
I think it is a bug to use `set` here. https://github.com/line/armeria/issues/3932#issuecomment-970254885
Could you use `add` for `header` and `headers` instead.